fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccrualBasedAccountingProcessorForLoan.java

```diff
@@ -129,15 +129,15 @@ private void createJournalEntriesForChargeAdjustment(LoanDTO loanDTO, LoanTransa
         Map<GLAccount, BigDecimal> accountMap = new HashMap<>();
 
         // handle principal payment (and reversals)
-        if (principalAmount != null && !(principalAmount.compareTo(BigDecimal.ZERO) == 0)) {
+        if (principalAmount != null && principalAmount.compareTo(BigDecimal.ZERO) != 0) {
```

Comment:
sonar?

Reply:
It should never happen, but still > 0 would be better... only positive amount it can be